src/reducers/weather-reducers.js

```diff
@@ -0,0 +1,21 @@
+import { FETCH_WEATHER } from "../actions";
+
+const weatherReducer = function (state = {}, action) {
+  switch (action.type) {
+    case FETCH_WEATHER:
+      return action.payload.data.map (function (city) {
```

Comment:
You're looking at the wrong key - you want `data.list` - as things are, this code breaks your app